BAZProjectC1/Container/ContainerViewController.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  ContainerViewController.swift
+//  BAZProjectC1
+//
+//  Created by 1054812 on 21/09/22.
+//
+
+import UIKit
+
+final class ContainerViewController: UIViewController {
+    
+    private let menuVC = FilterViewController()
+    private let homeVC = HomeAppViewController()
+    private var navVC: UINavigationController?
+    
+    private enum MenuState {
+        case opened
+        case closed
+    }
+    
+    private var menuState: MenuState = .closed
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.view.backgroundColor = .white
+        self.addChilds()
+    }
+    
+    private func addChilds() {
+        let menuNav = UINavigationController(rootViewController: menuVC)
+        self.menuVC.delegate = self
+        menuNav.navigationBar.barStyle = .default
+        self.addChild(menuNav)
+        menuNav.view.frame = self.view.frame
+        self.view.addSubview(menuNav.view)
+        menuNav.didMove(toParent: self)
+        
+        self.homeVC.delegate = self
+        let homeNav = UINavigationController(rootViewController: self.homeVC)
+        
+        self.addChild(homeNav)
+        self.view.addSubview(homeNav.view)
+        homeNav.didMove(toParent: self)
+        self.navVC = homeNav
+    }
+    
+    private func toggleMenu(completion: (() -> Void)?) {
+        switch menuState {
+        case .closed:
+            UIView.animate(withDuration: 0.5, delay: 0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0, options: .curveEaseInOut) {
+                self.navVC?.view.frame.origin.x = self.homeVC.view.frame.size.width - 100
+            } completion: { [weak self] done in
```

Comment:
Este `completion` no necesita capturar `[weak self]`. Puedes usar self directamente. Eso depende del tipo de clousure. Los _non-escaping clousure_ no generan _retain cycles_, por lo que el uso de `weak` no es necesario.

Reply:
Aquí una referencia por si quieres entender mejor:
https://betterprogramming.pub/introduction-to-non-escaping-closures-in-swift-3d6e8c4c6350#:~:text=An%20escaping%20closure%20is%20a,the%20function%20body%20gets%20executed.

Reply:
fixed